src/input/dotgraph.rs

```diff
@@ -33,14 +33,70 @@ impl<Node: Idx, Label: Idx> InputCapabilities<Node> for DotGraphInput<Node, Labe
     type GraphInput = DotGraph<Node, Label>;
 }
 
-pub struct DotGraph<Node: Idx, Label: Idx> {
-    node_count: Node,
-    edge_count: Node,
-    labels: Vec<Label>,
-    edges: EdgeList<Node>,
-    max_degree: Node,
-    max_label: Label,
-    label_frequency: HashMap<Label, usize>,
+pub struct DotGraph<Node, Label>
+where
+    Node: Idx,
+    Label: Idx,
+{
+    pub(crate) labels: Vec<Label>,
+    pub(crate) edge_list: EdgeList<Node>,
+    pub(crate) max_degree: Node,
+    pub(crate) max_label: Label,
+    pub(crate) label_frequency: HashMap<Label, usize>,
+}
+
+impl<Node, Label> DotGraph<Node, Label>
+where
+    Node: Idx,
+    Label: Idx + Hash,
+{
+    fn node_count(&self) -> Node {
+        Node::new(self.labels.len())
+    }
+
+    pub(crate) fn label_count(&self) -> usize {
```

Comment:
Not sure I understand the logic of this one. Maybe a comment could be nice? :relaxed:

Reply:
Mh... not sure what I thought while writing this. It should never be the case that the length of the map is larger than the maximum label. The other way around is true, e.g. you can have labels, `42` and `1337` and the `frequencies.len()` is `2`. We however need `1338` to correctly initialize the offsets for the label index csr. I simplified it.